core/capabilities/ccip/launcher/deployment.go

```diff
@@ -1,142 +1,45 @@
 		return nil
 	}
 	var err error
 
-	// shutdown active commit instance.
-	if c.commit.active != nil {
-		err = multierr.Append(err, c.commit.active.Close())
-	}
-
-	// shutdown candidate commit instance.
-	if c.commit.candidate != nil {
-		err = multierr.Append(err, c.commit.candidate.Close())
-	}
-
-	// shutdown active exec instance.
-	if c.exec.active != nil {
-		err = multierr.Append(err, c.exec.active.Close())
-	}
-
-	// shutdown candidate exec instance.
-	if c.exec.candidate != nil {
-		err = multierr.Append(err, c.exec.candidate.Close())
+	for _, oracle := range *c {
+		err = multierr.Append(err, (*oracle).Close())
 	}
 
 	return err
 }
 
-// StartActive starts the active OCR instances.
-func (c *ccipDeployment) StartActive() error {
+// Transition manages starting and stopping ocr instances
+// If there are any new config digests, we need to start those instances
+// If any of the previous config digests are no longer present, we need to shut those down
+// We don't care about if they're exec/commit or active/candidate, that all happens in the plugin
+func (c *ccipPlugins) Transition(prevPlugins *ccipPlugins) error {
 	var err error
-
-	err = multierr.Append(err, c.commit.active.Start())
-	err = multierr.Append(err, c.exec.active.Start())
-
-	return err
-}
-
-// CloseActive shuts down the active OCR instances.
-func (c *ccipDeployment) CloseActive() error {
-	var err error
-
-	err = multierr.Append(err, c.commit.active.Close())
-	err = multierr.Append(err, c.exec.active.Close())
-
-	return err
-}
-
-// TransitionDeployment handles the active-candidate deployment transition.
-// prevDeployment is the previous deployment state.
-// there are two possible cases:
-//
-// 1. both active and candidate are present in prevDeployment, but only active is present in c.
-// this is a promotion of candidate to active, so we need to shut down the active deployment
-// and make candidate the new active. In this case candidate is already running, so there's no
-// need to start it. However, we need to shut down the active deployment.
-//
-// 2. only active is present in prevDeployment, both active and candidate are present in c.
-// In this case, active is already running, so there's no need to start it. We need to
-// start candidate.
-func (c *ccipDeployment) TransitionDeployment(prevDeployment *ccipDeployment) error {
-	if prevDeployment == nil {
-		return fmt.Errorf("previous deployment is nil")
-	}
-
-	var err error
-	if prevDeployment.commit.candidate != nil && c.commit.candidate == nil {
-		err = multierr.Append(err, prevDeployment.commit.active.Close())
-	} else if prevDeployment.commit.candidate == nil && c.commit.candidate != nil {
-		err = multierr.Append(err, c.commit.candidate.Start())
-	} else {
-		return fmt.Errorf("invalid active-candidate deployment transition")
+	for _, digest := range maps.Keys(*prevPlugins) {
+		if *c == nil || (*c)[digest] == nil {
+			err = multierr.Append(err, (*(*prevPlugins)[digest]).Close())
+		}
```

Comment:
Lets add a comment here saying qualitatively what this does: shuts down instances that are no longer needed.

Reply:
Also can remove all the pointer dereferencing that is no longer needed.